### ISSUE TITLE
Multi-user version of Cherry Tree

### ISSUE THREAD

**Issue by edroberts1943** [giuspen/cherrytree#1606]
Multiuser version of Cherry Tree  (CTMU)

The following are FUNCTIONAL requirements for a multi-user version of the Cherry Tree note-taking application"

General Requirements:
• Supports up to N users.  In my case N = 4 would be sufficient.  Others may benefit from a much larger number.
• All users operate CTMU from a single ctb file. This may be on a computer remote for some users.
• Each user may specify a different location for backup ctb files to be stored.
• The implementation of Private nodes (see below) is not intended for security purposes, it is for convenience. Absolute protection is not a requirement.
• CTMU is backward compatible. It can open and initialize from existing non-multiuser Cherry Tree ctb files.
• There is no requirement to save a CTMU file as a non-multiuser file.

Node Requirements:
• Nodes are classified in 3 categories for the purposes of access control:
          Open, read/write for all users.
          Read Only, can be read/viewed by all users.  Only the creator of the node, the node owner, may write to the node.
          Private, only the node owner, the person who created the node, may view or write to the node.
• If a node's parent node is Private, that node (and all of its subnodes) will be Private.
• if a node's parent node is Read-Only, that node may be Open, Read Only, or Private.
• If a node's parent node is Open, that node may be Open, Read Only, or Private.
• Node properties that affect the way data is saved can be changed only by the node's owner.
• Any users can specify how a node will be displayed independently of all other users.  This affects only that user.

Backup Requirements:
• Backup cbt files will be the same for all users. Private nodes will be included in all backups (even though a particular user may not be able to see these).

**Comment by Insvl1129** [comment]
I don't think it is necessary. Refer to major software. This is a personal note-taking software, not a commercial office spreadsheet software.
